### PR TITLE
Core: Fix NPE for parent snapshot does not exist

### PR DESCRIPTION
**Pull request by hililiwei**
Fix NPE for parent snapshot does not exist

### CODE REVIEW

File: core/src/main/java/org/apache/iceberg/util/SnapshotUtil.java [hunk 1]
```diff
@@ -72,7 +72,7 @@ public static boolean isAncestorOf(Table table, long ancestorSnapshotId) {
   public static boolean isParentAncestorOf(
       Table table, long snapshotId, long ancestorParentSnapshotId) {
     for (Snapshot snapshot : ancestorsOf(snapshotId, table::snapshot)) {
-      if (snapshot.parentId() == ancestorParentSnapshotId) {
+      if (snapshot.parentId() != null && snapshot.parentId() == ancestorParentSnapshotId) {
```
Comment (ajantha-bhat) [inline core/src/main/java/org/apache/iceberg/util/SnapshotUtil.java:75]: Can we add a test case for this scenario as well?

Comment (hililiwei) [reply core/src/main/java/org/apache/iceberg/util/SnapshotUtil.java:75]: SnapshotUtil doesn't have a test case class for it. So I created one, adding some additional test cases.

**Review by Fokko — APPROVED** [review]

**Review by aokolnychyi — APPROVED** [review]

**Comment by aokolnychyi** [comment]
Thanks, @hililiwei! Thanks for reviewing, @ajantha-bhat @Fokko!

**Comment by aokolnychyi** [comment]
@hililiwei, looks like we started getting test failures. Could you check? I will be able to check in 2 hours as well.

```
TestSnapshotUtil > oldestAncestor FAILED
[367](https://github.com/apache/iceberg/actions/runs/3277597635/jobs/5395173940#step:6:368)
    java.lang.AssertionError: expected:<2> but was:<1>
[368](https://github.com/apache/iceberg/actions/runs/3277597635/jobs/5395173940#step:6:369)
        at org.junit.Assert.fail(Assert.java:89)
[369](https://github.com/apache/iceberg/actions/runs/3277597635/jobs/5395173940#step:6:370)
        at org.junit.Assert.failNotEquals(Assert.java:835)
[370](https://github.com/apache/iceberg/actions/runs/3277597635/jobs/5395173940#step:6:371)
        at org.junit.Assert.assertEquals(Assert.java:647)
[371](https://github.com/apache/iceberg/actions/runs/3277597635/jobs/5395173940#step:6:372)
        at org.junit.Assert.assertEquals(Assert.java:633)
[372](https://github.com/apache/iceberg/actions/runs/3277597635/jobs/5395173940#step:6:373)
        at org.apache.iceberg.util.TestSnapshotUtil.oldestAncestor(TestSnapshotUtil.java:133)
```

**Pull request merged**